src/components/SignInPageForm/index.js

```diff
@@ -10,6 +16,22 @@ class Form extends React.Component {
         // Password Managers need these attributes to be able to identify the form elements properly.
         this.form.setAttribute('method', 'post');
         this.form.setAttribute('action', '/');
+
+        this.form.addEventListener('submit', this.preventFormDefault);
+    }
```

Comment:
I'm still confused, here we set post action. And we have a `submit` listener which 

> When enter is pressed form is submitted to action url (POST /).
> As we are using controlled component, we need to disable it here.

Reply:
@Santhosh-Sellavel this is the default browser behaviour to submit form data to action url, and this makes a full page refresh. As in our case this is a web app and does not require refresh. We need to prevent this default behaviour, otherwise user will be redirected to / when he presses enter on password input field. Form submition is manually handled using ajax request.

Reply:
`react-native-web` does not accept `onSubmit` prop, therefore `addEventListener` is used here.

Reply:
I'm okay here,  but double checking @thienlnam Any thoughts?

Reply:
> react-native-web does not accept onSubmit prop

Where did you find this out? I'm pretty sure this is not true as we have it many places and onSubmit is a common prop that seems like it would be supported in react-native-web

Reply:
I agree that this should be supported, and this has been raised earlier in this [issue](https://github.com/necolas/react-native-web/issues/1034), also you can check the docs page [here](https://necolas.github.io/react-native-web/docs/interactions/) this mentions the supported props.

Reply:
Took a look, it seems like in that issue they are trying to add a onSubmit handler to the View which isn't even supported in react-native.  The on submit has to be applied to the button event

Following the code, it seems that we're setting up our own onSubmit event handler in FormSubmit which wraps is wrapped in all the [sign in content](https://github.com/Expensify/App/blob/ba7c3812ed0ef5b5a41ec30f32de18c0c180ac4b/src/pages/SetPasswordPage.js#L98-L124)

Can you try preventing default in the FormSubmit component?

Reply:
I tried, but it does not work.